File: 2025-09-16 - LstDupEncadeada/Exemplo_Python/TesteListaDuplaEncadeada.py
# ------------------------------------------------------------------------
# Programa principal de teste da ListaDuplaEncadeada
# ------------------------------------------------------------------------
from ListaDuplaEncadeada import ListaDuplaEncadeada

def main() -> None:
   lista = ListaDuplaEncadeada()

   print("== Inserções ==")
   lista.InserirNoInicio("INF101", "Algoritmos", 60)
   lista.InserirNoFim("INF102", "Estruturas de Dados", 80)
   lista.InserirNoFim("INF103", "Banco de Dados", 70)
   lista.ImprimirLista()

   print("\n== Inserir em posições específicas ==")
   lista.InserirNaPosicao("INF150", "Lógica", 40, 1)
   lista.InserirNaPosicao("INF100", "Introdução", 30, 0)
   lista.InserirNaPosicao("INF200", "Redes", 50, lista.Tamanho())
   lista.ImprimirLista()

   print("\n== Atualização e remoção ==")
   print("Atualizar INF103:", lista.AtualizarDisciplina("INF103", "Banco Avançado", 90))
   print("Remover INF100:", lista.RemoverPorCodigo("INF100"))
   lista.ImprimirLista()

if __name__ == "__main__":
   main()

File: 2025-09-16 - Lista Dup Encadeada/Exemplo_C#/TesteListaDuplaEncadeada.cs
// ------------------------------------------------------------------------
// Programa principal para demonstrar todas as operações
// ------------------------------------------------------------------------

using System;
using Estruturas;

namespace Estruturas
{
   class Program
   {
      static void Main(string[] args)
      {
         ListaDuplaEncadeada lista = new ListaDuplaEncadeada();

         Console.WriteLine("== Inserções ==");
         lista.InserirNoInicio("INF101", "Algoritmos", 60);
         lista.InserirNoFim("INF102", "Estruturas de Dados", 80);
         lista.InserirNoFim("INF103", "Banco de Dados", 70);
         lista.ImprimirLista();

         Console.WriteLine("\n== Inserir em posições específicas ==");
         lista.InserirNaPosicao("INF150", "Lógica", 40, 1);
         lista.InserirNaPosicao("INF100", "Introdução", 30, 0);
         lista.InserirNaPosicao("INF200", "Redes", 50, lista.Tamanho());
         lista.ImprimirLista();

         Console.WriteLine("\n== Atualização e remoção ==");
         Console.WriteLine("Atualizar INF103: " +
            lista.AtualizarDisciplina("INF103", "Banco Avançado", 90));
         Console.WriteLine("Remover INF100: " + lista.RemoverPorCodigo("INF100"));
         lista.ImprimirLista();
      }
   }
}
